101-bastion-hub-spoke-vnet/azuredeploy.json

```diff
@@ -0,0 +1,298 @@
+                    "properties": {
+                        "addressPrefix": "[variables('subnetBastionPrefix')]"
+                    }
+                    
+                }
+            ]
+        }
+    },
+    {
+        "name": "[parameters('vNetSpokeName')]",
+        "type": "Microsoft.Network/virtualNetworks",
+        "apiVersion": "2020-05-01",
+        "location": "[parameters('location')]",
+        "properties": {
+            "addressSpace": {
+                "addressPrefixes": [
+                    "[variables('vNetSpokePrefix')]"
+                ]
+            },
+            "subnets": [
+                {
+                    "name": "[variables('subnetSpokeName')]",
+                    "properties": {
+                        "addressPrefix": "[variables('subnetSpokePrefix')]"
+                    }
+                }
+            ]
+        }
+    },
+    {
+        "name": "[concat(parameters('vNetHubName'),'/peering-to-',parameters('vNetSpokeName'))]",
+        "type": "Microsoft.Network/virtualNetworks/virtualNetworkPeerings",
+        "apiVersion": "2020-05-01",
+        "dependsOn": [
+            "[resourceId('Microsoft.Network/virtualNetworks', parameters('vNetHubName'))]",
+            "[resourceId('Microsoft.Network/virtualNetworks', parameters('vNetSpokeName'))]"
+        ],
+        "properties": {
+            "allowVirtualNetworkAccess": true,
+            "allowForwardedTraffic": false,
+            "allowGatewayTransit": false,
+            "useRemoteGateways": false,
+            "remoteVirtualNetwork": {
+                "id": "[resourceId('Microsoft.Network/virtualNetworks',parameters('vNetSpokeName'))]"
+            }
+        }
+    },
+    {
+        "name": "[concat(parameters('vNetSpokeName'),'/peering-to-',parameters('vNetHubName'))]",
+        "type": "Microsoft.Network/virtualNetworks/virtualNetworkPeerings",
+        "apiVersion": "2020-05-01",
+        "dependsOn": [
+            "[resourceId('Microsoft.Network/virtualNetworks', parameters('vNetHubName'))]",
+            "[resourceId('Microsoft.Network/virtualNetworks', parameters('vNetSpokeName'))]"
+        ],        
+        "properties": {
+            "allowVirtualNetworkAccess": true,
+            "allowForwardedTraffic": false,
+            "allowGatewayTransit": false,
+            "useRemoteGateways": false,
+            "remoteVirtualNetwork": {
+                "id": "[resourceId('Microsoft.Network/virtualNetworks',parameters('vNetHubName'))]"
+            }
+        }
+    },
+    {
+        "name": "[variables('nsgName')]",
+        "type": "Microsoft.Network/networkSecurityGroups",
+        "apiVersion": "2020-05-01",
+        "location": "[parameters('location')]",
+        "properties": {
+            "securityRules": [
+                {
+                    "name": "nsgRule1",
+                    "properties": {
+                        "description": "description",
```

Comment:
using * and *  what's this trying to secure?  (should also give a proper description)

Reply:
I updated this to refer to a specific port. I also added a Public IP to the VM so that it makes more sense to open the port.